### PR TITLE
Minimize the amount of rendering done when the user hovers over the console (#1743)

### PR DESCRIPTION
**Pull request by hbenl**

### CODE REVIEW

File: src/devtools/client/webconsole/components/Output/Message.js [hunk 1]
```diff
@@ -17,17 +17,6 @@ const CollapseButton = require("devtools/client/webconsole/components/Output/Col
...
-  if (!message?.frame || !hoveredPoint?.location) {
-    return false;
-  }
-
-  const keyOne = getLocationKey(hoveredPoint.location);
-  const keyTwo = getLocationKey(message.frame);
-  return keyOne == keyTwo;
-}
 
 class Message extends Component {
```
Comment (jasonLaster) [inline src/devtools/client/webconsole/components/Output/Message.js:21]: I wonder if we can promote Message to a `PureComponent` and drop `shouldComponentUpdate`

https://reactjs.org/docs/react-api.html#reactpurecomponent

File: src/devtools/client/webconsole/components/Output/MessageContainer.js [hunk 2]
```diff
@@ -61,14 +63,16 @@ class MessageContainer extends Component {
...
+      "isSecondaryHighlighted",
+      "shouldScrollIntoView",
     ];
 
     return triggeringUpdateProps.some(prop => this.props[prop] !== nextProps[prop]);
   }
 
   render() {
-    const message = this.props.getMessage();
+    const message = this.props.message;
```
Comment (jasonLaster) [inline src/devtools/client/webconsole/components/Output/MessageContainer.js:75]: `const {message} = this.props;

**Review by jasonLaster — APPROVED** [review]
Looks great

File: src/devtools/client/webconsole/components/Output/Message.js (hunk 1 above)
Comment (hbenl) [reply src/devtools/client/webconsole/components/Output/Message.js:21]: I was wondering that too, but I decided to try with minimal changes for now. We're passing a lot of props to the message components that I think may trigger unnecessary renders. But we should revisit this when we refactor the code.

File: src/ui/components/Timeline/index.tsx [hunk 3]
```diff
@@ -184,20 +209,26 @@ class Timeline extends Component<PropsFromRedux> {
...
+              point={point.point}
+              time={point.time}
+              hasFrames={!!point.frame}
+              location={point.frame?.[0]}
+              currentTime={currentTime}
+              isPrimaryHighlighted={isPrimaryHighlighted}
+              isSecondaryHighlighted={isSecondaryHighlighted}
+              zoomRegion={zoomRegion}
+              overlayWidth={this.overlayWidth}
+              onSeek={() => {}}
```
Comment (jasonLaster) [inline src/ui/components/Timeline/index.tsx:228]: this seems dangerous for re-renders. But i assume it is okay

Comment (hbenl) [reply src/ui/components/Timeline/index.tsx:228]: Dangerous how?

Comment (jasonLaster) [reply src/ui/components/Timeline/index.tsx:228]: its a new object everytime, so could trigger a re-render

Comment (hbenl) [reply src/ui/components/Timeline/index.tsx:228]: Ah, yes, we should definitely fix this before making `Marker` a `PureComponent`.

**Pull request merged**